src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+    }
+
+    /**
+     * Set the task as done and prints the information
+     *
+     * @param orderStr Order of the task inside the list
+     */
+    private static void setTaskDone(String orderStr) {
+        int order;
+        if (orderStr == null) {
+            System.out.println("Oh, no! You didn't specify which task you are done with!");
+            return;
+        }
+        try {
+            order = Integer.parseInt(orderStr);
+        } catch (NumberFormatException nfe) {
+            System.out.println("Oh, no! I don't get what you are trying to say!");
+            return; //orderStr is not an integer
+        }
+        int index = order - 1;
+        if (index >= 0 && index < listCount) {
+            if (tasks[index].isDone()) {
+                System.out.println("Mama Mia! You have already done this task~");
+            } else {
+                tasks[index].markAsDone();
+                System.out.println("We did it! Good job little guy.");
+            }
+            System.out.println("\t" + tasks[index].toString());
+        } else {
+            System.out.println("Oh, no! There is no such task!");
+        }
+    }
+
+    /**
+     * Prints the entire list of tasks
+     */
+    private static void printList() {
+        int doneCount = 0;
+        System.out.println("Here we go! These are the tasks you have:");
+        for (int i = 0; i < listCount; i++) {
+            Task task = tasks[i];
+            int order = i + 1;
+            if (task.isDone) {
+                doneCount++;
+            }
+            System.out.println(String.format("\t%d. %s", order, task.toString()));
+        }
+        System.out.println(String.format("You have %d tasks and you completed %d of them", listCount, doneCount));
+    }
+
+
+    /**
+     * Adds new task to the list
+     *
+     * @param type Type of the new task
+     * @param args Arguments that are passed with the type
+     */
+    public static void addTask(Task.Type type, String args) {
+        switch (type) {
+        case DEADLINE:
+            tasks[listCount] = createDeadline(args);
+            break;
+        case TODO:
+            tasks[listCount] = createTodo(args);
+            break;
+        case EVENT:
+            tasks[listCount] = createEvent(args);
+            break;
+        }
+        if (tasks[listCount] != null) {
+            System.out.println("Okey Dokey! Added: " + tasks[listCount++]);
+        } else {
+            System.out.println("Oh, no! A valid input please~");
+        }
+    }
+
+    /**
+     * Creates a new deadline task
+     *
+     * @param args Arguments of the command
+     * @return Newly created deadline
+     */
+    public static Deadline createDeadline(String args) {
+        if (args == null) {
+            return null;
+        }
+        String[] argArr = args.split("/by");
+        if (argArr.length != 2) {
+            return null;
+        }
+        String name = argArr[0];
+        String by = argArr[1];
```

Comment:
Im not too sure what the 0 and 1 is referring to in argArr array. Perhaps refactoring away this magic number might improve code readability.